omc3/optics_measurements/phase.py

```diff
@@ -151,33 +183,37 @@ def _calculate_with_compensation(
     LOGGER.info("Calculating phase advances")
     LOGGER.info(f"Measured tune in plane {plane} = {tunes[plane]['Q']}")
 
-    df = model_df.loc[:, ["S", f"MU{plane}"]]
+    df = model_df.loc[:, [S, f"{PHASE_ADV}{plane}"]]
     how = 'outer' if meas_input.union else 'inner'
 
     dpp_value = meas_input.analyse_dpp
     if dpp_value is None:
         check_and_warn_about_offmomentum_data(input_files, plane, id_="Phase calculations")
 
-    df = pd.merge(df, input_files.joined_frame(plane, [f"MU{plane}", f"{ERR}MU{plane}"],
-                                               dpp_value=dpp_value, how=how),
-                  how='inner', left_index=True, right_index=True)
-    df[input_files.get_columns(df, f"MU{plane}")] = (input_files.get_data(df, f"MU{plane}")
-                                                     * meas_input.accelerator.beam_direction
-                                                     )
-    phases_mdl = df.loc[:, f"MU{plane}"].to_numpy()
-    phase_advances = {"MODEL": _get_square_data_frame(_get_all_phase_diff(phases_mdl), df.index)}
-    if compensation == "model":
+    joined_df = input_files.joined_frame(plane, [f"{PHASE_ADV}{plane}", f"{ERR}{PHASE_ADV}{plane}"], dpp_value=dpp_value, how=how)
```

Comment:
It was never so clear to me what exactly comes out of `input_files.joined_frame` (docstrings there did not help). Could we add a comment or change the variable name to make this a bit more explicit?

Reply:
added more info to the docstring of `joined_frame`. 

To be honest, it would not have been my first choice in how to implement this, but it does its job and would be a lot of work to rethink and rewrite